internal/cmd/asyncapi/command_import.go

```diff
@@ -0,0 +1,391 @@
+	c.Flags().String("schema-registry-api-secret", "", "API secret for Schema Registry.")
+	return c.Command
+}
+
+func getFlagsReverse(cmd *cobra.Command) (*flagsReverse, error) {
+	overwrite, err := cmd.Flags().GetBool("overwrite")
+	if err != nil {
+		return nil, err
+	}
+	kafkaApiKey, err := cmd.Flags().GetString("kafka-api-key")
+	if err != nil {
+		return nil, err
+	}
+	schemaRegistryApiKey, err := cmd.Flags().GetString("schema-registry-api-key")
+	if err != nil {
+		return nil, err
+	}
+	schemaRegistryApiSecret, err := cmd.Flags().GetString("schema-registry-api-secret")
+	if err != nil {
+		return nil, err
+	}
+
+	return &flagsReverse{
+		overwrite:               overwrite,
+		kafkaApiKey:             kafkaApiKey,
+		schemaRegistryApiKey:    schemaRegistryApiKey,
+		schemaRegistryApiSecret: schemaRegistryApiSecret,
+	}, nil
+}
+
+func (c *command) reverse(cmd *cobra.Command, strings []string) error {
+	if len(os.Args) < 4 {
+		err := errors.New("Provide Input spec filename as argument.")
+		return err
+	}
+	fileName := os.Args[3]
+	asyncSpec, err := os.ReadFile(fileName)
+	if err != nil {
+		return err
+	}
+	spec := new(Async)
+	err = yaml.Unmarshal(asyncSpec, &spec)
+	if err != nil {
+		return fmt.Errorf("unable to unmarshal yaml file: %v", err)
+	}
+	//Get Flags
+	flagsRev, err := getFlagsReverse(cmd)
+	if err != nil {
+		return err
+	}
+	//Getting Kafka Cluster & SR
+	flags := new(flags)
+	flags.kafkaApiKey = flagsRev.kafkaApiKey
+	flags.schemaRegistryApiKey = flagsRev.schemaRegistryApiKey
+	flags.schemaRegistryApiSecret = flagsRev.schemaRegistryApiSecret
+	details, err := c.getAccountDetails(flags)
+	if err != nil {
+		return err
+	}
+	//Add Topics
+	for topicName, topicDetails := range spec.Channels {
+		//For each topic in spec
+		//Reset topicExists variable
+		topicExists = false
+		topicNotCreated = false
+		output.Println("Importing topic: ", topicName)
+		err := addTopic(details, topicName, topicDetails.TopicBinding.Kafka, flagsRev.overwrite)
+		if err != nil {
+			//unable to create kafka topic
+			log.CliLogger.Debugf("unable to create kafka topic: %v", err)
+			topicNotCreated = true
+		}
+		//If topic exists and overwrite flag is false, move to the next channel in spec
+		if topicExists && !flagsRev.overwrite {
+			continue
+		}
+		//Register schema
+		schemaId, err := registerSchema(details, topicName, spec.Components)
+		if err != nil {
+			log.CliLogger.Warn(err)
+		}
+		//Update subject compatibility
+		err = updateSubjectCompatibility(details, spec.Channels[topicName].XMessageCompatibility, topicName+"-value")
+		if err != nil {
+			log.CliLogger.Warn(err)
+		}
+		//Add tags
+		_, _, err = addSchemaTags(details, spec.Components, topicName, schemaId)
+		if err != nil {
+			log.CliLogger.Warn(err)
+		}
+		if topicNotCreated == false && spec.Channels != nil {
+			_, _, err = addTopicTags(details, spec.Channels[topicName].Subscribe, topicName)
+			if err != nil {
```

Comment:
Shrinking the scope of your variables will help prevent bugs in the future.
```suggestion
			if _, _, err := addTopicTags(details, spec.Channels[topicName].Subscribe, topicName); err != nil {
```